### PR TITLE
5.7 version of user filtering for audit log

### PR DESCRIPTION
**Pull request by gl-sergei**
Jenkins: http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/218/

### CODE REVIEW

**Comment by gl-sergei** [comment]
Jenkins: http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/224/
Merge with some changes due to slightly different API and additional PSI instrumentation.
Commit 033c825 is an example of how we can use `security_context_service` here to filter against accounts from `mysql.user`. I am not sure if we must use it. Oracle's plugin does.

**Comment by gl-sergei** [comment]
oops, forgot to move new test case into `plugin/audit_log` dir

File: plugin/audit_log/audit_log.c [hunk 1]
```diff
@@ -787,6 +802,42 @@ int is_event_class_allowed_by_policy(mysql_event_class_t class,
...
+              audit_log_exclude_accounts == NULL);
+
+  if (event_class == MYSQL_AUDIT_CONNECTION_CLASS)
+  {
+    LEX_STRING priv_user, priv_host;
+    MYSQL_SECURITY_CONTEXT ctx;
+    my_svc_bool ret;
+
+    ret= thd_get_security_context(thd, &ctx);
+    DBUG_ASSERT(!ret);
```
Comment (laurynas-biveinis) [inline plugin/audit_log/audit_log.c]: IMHO any potential security context API failure handling must be much more defensive. If any of those calls fails, further processing should be aborted with a my_message

Comment (gl-sergei) [inline plugin/audit_log/audit_log.c]: but can it really fail? `thd_get_security_context` is just a `thd->security_context()` and in most places in server it used like this:

```
...
thd->security_context()->host_or_ip().str,
thd->security_context()->priv_user().str,
...
```

Comment (laurynas-biveinis) [inline plugin/audit_log/audit_log.c]: We don't know how its implementation is going to change in the future

**Comment by gl-sergei** [comment]
I just updated the last commit with error handling.

**Comment by laurynas-biveinis** [comment]
LGTM

**Comment by laurynas-biveinis** [comment]
This also needs a 5.7 blueprint

**Comment by gl-sergei** [comment]
I realised it couple of days ago and created this: https://blueprints.launchpad.net/percona-server/+spec/audit-filter-users-5.7

**Pull request merged**